### PR TITLE
WT-10580 Rename object_prefix and prefix in Azure Connection Class

### PR DESCRIPTION
**Pull request by lting0804**
- Updated obj_prefix to bucket_prefix 
- Amended unit tests to account for this change

### CODE REVIEW

File: ext/storage_sources/azure_store/test/test_azure_connection.cpp [hunk 1]
```diff
@@ -117,28 +118,28 @@ TEST_CASE("Testing Azure Connection Class", "azure-connection")
...
+        REQUIRE(conn.list_objects(object_name, objects, false) == 0);
         REQUIRE(objects.size() == blob_objects.size());
         objects.clear();
 
         // List single. Object size should be 1.
-        REQUIRE(conn.list_objects(obj_prefix + object_name, objects, true) == 0);
+        REQUIRE(conn.list_objects(object_name, objects, true) == 0);
         REQUIRE(objects.size() == 1);
-
+
```
Comment (jiechenbo) [inline ext/storage_sources/azure_store/test/test_azure_connection.cpp:136]: np: added whitespace

File: ext/storage_sources/azure_store/test/test_azure_connection.cpp [hunk 2]
```diff
@@ -73,9 +73,10 @@ TEST_CASE("Testing Azure Connection Class", "azure-connection")
       std::getenv("AZURE_STORAGE_CONNECTION_STRING"), "myblobcontainer1");
     bool exists = false;
 
-    std::string obj_prefix = randomize_test_prefix();
+    std::string bucket_prefix = randomize_test_prefix();
 
-    azure_connection conn = azure_connection("myblobcontainer1", obj_prefix);
+    azure_connection conn = azure_connection("myblobcontainer1", bucket_prefix);
+    azure_connection conn_second = azure_connection("myblobcontainer1", "pfx_");
```
Comment (jiechenbo) [inline ext/storage_sources/azure_store/test/test_azure_connection.cpp:79]: I don't think we need the second connection class here.

**Review by jiechenbo — APPROVED** [review]
LGTM! just two np comments

File: ext/storage_sources/azure_store/test/test_azure_connection.cpp (hunk 2 above)
Comment (lting0804) [reply ext/storage_sources/azure_store/test/test_azure_connection.cpp:79]: yup removed :)

File: ext/storage_sources/azure_store/test/test_azure_connection.cpp (hunk 1 above)
Comment (lting0804) [reply ext/storage_sources/azure_store/test/test_azure_connection.cpp]: removed

**Review by korteland — APPROVED** [review]
LGTM!

**Comment by lting0804** [comment]
evergreen merge

**Comment by lting0804** [comment]
evergreen merge

**Pull request merged**